File: tests/inputs/hello.c
/* Sample C file to exercise syntax highlighting, indents,
 * and folds in vorto. Open with `vorto assets/samples/hello.c`. */

#include <stdio.h>
#include <string.h>

#define MAX_TAGS 8

typedef enum {
    NEGATIVE,
    ZERO,
    POSITIVE_EVEN,
    POSITIVE_ODD,
} Classification;

typedef struct {
    const char *name;
    int age;
    const char *tags[MAX_TAGS];
    int tag_count;
} Person;

static const char *classify_name(Classification c) {
    switch (c) {
        case NEGATIVE:      return "negative";
        case ZERO:          return "zero";
        case POSITIVE_EVEN: return "positive even";
        default:            return "positive odd";
    }
}

static Classification classify(int n) {
    if (n < 0) return NEGATIVE;
    if (n == 0) return ZERO;
    return (n % 2 == 0) ? POSITIVE_EVEN : POSITIVE_ODD;
}

int main(void) {
    Person alice = {
        .name = "Alice",
        .age = 30,
        .tags = {"admin", "early_bird"},
        .tag_count = 2,
    };

    printf("Hello, %s!\n", alice.name);

    for (int n = -2; n <= 5; n++) {
        printf("%d -> %s\n", n, classify_name(classify(n)));
    }

    return 0;
}

File: tests/inputs/hello.py
# Sample Python file to exercise syntax highlighting, indents,
# and folds in vorto. Open with `vorto assets/samples/hello.py`.

from __future__ import annotations

from dataclasses import dataclass, field
from enum import Enum


class Classification(Enum):
    NEGATIVE = "negative"
    ZERO = "zero"
    POSITIVE_EVEN = "positive even"
    POSITIVE_ODD = "positive odd"


@dataclass
class Person:
    """A person with a name, age, and free-form tags."""

    name: str
    age: int
    tags: list[str] = field(default_factory=list)

    @property
    def is_adult(self) -> bool:
        return self.age >= 18


def greet(person: Person, prefix: str = "Hello") -> str:
    return f"{prefix}, {person.name}!"


def classify(n: int) -> Classification:
    if n < 0:
        return Classification.NEGATIVE
    if n == 0:
        return Classification.ZERO
    return Classification.POSITIVE_EVEN if n % 2 == 0 else Classification.POSITIVE_ODD


def squares(numbers: list[int]) -> list[int]:
    return [x * x for x in numbers if x * x > 5]


def main() -> None:
    alice = Person(name="Alice", age=30, tags=["admin", "early_bird"])
    print(greet(alice))

    for n in range(-2, 6):
        print(f"{n} -> {classify(n).value}")

    print("squares:", squares([1, 2, 3, 4, 5]))


if __name__ == "__main__":
    main()
